lib/puppet/resource/catalog.rb

```diff
@@ -366,18 +369,14 @@ def resource(type, title = nil)
       # an instance has to be created in order to construct the unique key used when
       # searching for aliases, or when app_management is active and nothing is found in
       # which case it is needed by the CapabilityFinder.
-# PUP-4947
       res = nil
-#      app_mgnt = Puppet[:app_management]
-#      if app_mgnt || !@aliases.empty?
       res = Puppet::Resource.new(type, title, { :environment => @environment_instance })
 
-        # No need to build the uniqueness key unless there are aliases
-      result = @resource_table[[type_name, res.uniqueness_key].flatten] unless @aliases.empty?
-# PUP-4947
-#      end
```

Comment:
There are no comments about PUP-4947 afaict. Think you are commenting on the original where I removed those comments ??? PUP-4947 is only mentioned in a commit message as it states it cleans up commented out code from that ticket.